config.go

```diff
@@ -95,6 +104,8 @@ func (c *Config) Validate() error {
 	switch {
 	case c.Group.Offsets.Retry.Max < 0:
 		return sarama.ConfigurationError("Group.Offsets.Retry.Max must be >= 0")
+	case c.Group.Offsets.Synchronization.DwellTime == nil || *(c.Group.Offsets.Synchronization.DwellTime) > time.Minute*10:
```

Comment:
c.Group.Session.Timeout has to do with the heartbeat frequency. The only aspect of it that has to do with committing offsets is that after the session times out for a client (due to lack of heartbeat), a partition rebalance will occur. By that point, if the offsets had not been committed by the dead client, there's nothing that can be done.

Note that a dwell time greater than the session timeout is also reasonable. The backpressure scenario in the original description can occur regardless of session timeout; the client is heartbeating, just not reading in that scenario.

10 minutes is chosen as a maximum time that won't cause a "major outage" for most customers. 2x this time is 20 minutes. This is on the order of the time that someone responding to a pager notification would be able to respond with an investigation. Further, it seems unlikely that waiting more than 10 or 20 minutes for a committed offset would provide more value to the customer than simply resuming consumption, at the risk of some duplication of effort. Consumer checkpointing of offsets is usually far more granular than 10 minutes.